bootstrap/util/configowner_test.go

```diff
@@ -20,113 +20,134 @@ import (
+					},
+				},
+				Namespace: "my-ns",
+				Name:      "my-resource-owned-by-machine",
 			},
-		},
-		Namespace: "my-ns",
-		Name:      "my-resource-owned-by-machine",
-	}
-	configOwner, err := GetConfigOwner(context.TODO(), c, objm)
-	if err != nil {
-		t.Fatalf("did not expect an error but found one: %v", err)
-	}
-	if configOwner == nil {
-		t.Fatal("expected a configOwner but got nil")
-	}
-	if configOwner.ClusterName() != "my-cluster" {
-		t.Fatalf("did not expect ClusterName: %q", configOwner.ClusterName())
-	}
-	if !configOwner.IsInfrastructureReady() {
-		t.Fatalf("did not expect InfrastructureReady: %v", configOwner.IsInfrastructureReady())
-	}
-	if configOwner.DataSecretName() == nil {
-		t.Fatalf("did not expect DataSecretName: %v", configOwner.DataSecretName())
-	}
-	if !configOwner.IsControlPlaneMachine() {
-		t.Fatalf("did not expect IsControlPlane: %v", configOwner.IsControlPlaneMachine())
-	}
-}
+		}
+		configOwner, err := GetConfigOwner(context.TODO(), c, obj)
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(configOwner).ToNot(BeNil())
+		g.Expect(configOwner.ClusterName()).To(BeEquivalentTo("my-cluster"))
+		g.Expect(configOwner.IsInfrastructureReady()).To(BeTrue())
+		g.Expect(configOwner.IsControlPlaneMachine()).To(BeTrue())
+		g.Expect(*configOwner.DataSecretName()).To(BeEquivalentTo("my-data-secret"))
+	})
 
-func TestGetConfigOwnerNotFound(t *testing.T) {
-	scheme := runtime.NewScheme()
-	if err := clusterv1.AddToScheme(scheme); err != nil {
-		t.Fatal("failed to register cluster api objects to scheme")
-	}
+	t.Run("should get the owner when present (MachinePool)", func(t *testing.T) {
+		myPool := &clusterv1.MachinePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-machine-pool",
+				Namespace: "my-ns",
+				Labels: map[string]string{
+					clusterv1.MachineControlPlaneLabelName: "",
+				},
+			},
+			Spec: clusterv1.MachinePoolSpec{
+				ClusterName: "my-cluster",
+			},
+			Status: clusterv1.MachinePoolStatus{
+				InfrastructureReady: true,
+			},
+		}
 
-	c := fake.NewFakeClientWithScheme(scheme)
-	objm := metav1.ObjectMeta{
-		OwnerReferences: []metav1.OwnerReference{
-			{
-				Kind:       "Machine",
-				APIVersion: clusterv1.GroupVersion.String(),
-				Name:       "my-machine",
+		c := fake.NewFakeClientWithScheme(scheme, myPool)
+		obj := &bootstrapv1.KubeadmConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind:       "MachinePool",
+						APIVersion: clusterv1.GroupVersion.String(),
+						Name:       "my-machine-pool",
+					},
+				},
+				Namespace: "my-ns",
+				Name:      "my-resource-owned-by-machine-pool",
 			},
-		},
-		Namespace: "my-ns",
-		Name:      "my-resource-owned-by-machine",
-	}
-	_, err := GetConfigOwner(context.TODO(), c, objm)
-	if err == nil {
-		t.Fatal("expected error, got nil")
-	}
-}
+		}
+		configOwner, err := GetConfigOwner(context.TODO(), c, obj)
+		g.Expect(err).ToNot(HaveOccurred())
```

Comment:
nit: `NotTo` would read better here than `ToNot`
```suggestion
		g.Expect(err).NotTo(HaveOccurred())
```